coalib/results/Diff.py

```diff
@@ -351,14 +351,20 @@ def __bool__(self):
     def delete_line(self, line_nr):
         """
         Mark the given line nr as deleted. The first line is line number 1.
+
+        Raises an error if line nr doesn't exist in the diff.
         """
+        line_nr = 0 if (len(self._file) < line_nr) else line_nr
```

Comment:
Ah you're setting it to zero which will raise an `ValueError`.

Why not simply do a `raise ValueError` if it's out of bounds? Actually, do a `raise IndexError` since that makes more sense.